src/openms/source/ANALYSIS/OPENSWATH/IonMobilityScoring.cpp

```diff
@@ -430,27 +469,28 @@ namespace OpenMS
     scores.im_drift_weighted = computed_im_weighted;
 
     // Step 2: Align the IonMobilogram vectors to the grid
-    std::vector<double> im_grid = computeGrid_(mobilograms, eps);
-    std::vector< std::vector< double > > aligned_mobilograms;
-    for (const auto & mobilogram : mobilograms)
+    std::vector<double> im_grid = computeGrid_(ms2_mobilograms, eps);
+    std::vector< OpenMS::Mobilogram > aligned_ms2_mobilograms;
+    for (const auto & mobilogram : ms2_mobilograms)
     {
-      std::vector< double > arr_int, arr_IM;
+      Mobilogram aligned_mobilogram;
       Size max_peak_idx = 0;
-      alignToGrid_(mobilogram, im_grid, arr_int, arr_IM, eps, max_peak_idx);
-      if (!arr_int.empty()) aligned_mobilograms.push_back(arr_int);
+      alignToGrid_(mobilogram, im_grid, aligned_mobilogram, eps, max_peak_idx);
+      if (!aligned_mobilogram.empty()) aligned_ms2_mobilograms.push_back(aligned_mobilogram);
```

Comment:
I think move is also possible here
```suggestion
      if (!aligned_mobilogram.empty()) aligned_ms2_mobilograms.push_back(std::move(aligned_mobilogram));
```